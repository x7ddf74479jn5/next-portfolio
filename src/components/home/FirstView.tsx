import Image from "next/image";
import styles from "src/styles/components/home/FirstView.module.scss";

import { NextArrowButton } from "../common/NextArrowButton";

const FirstView = () => {
  return (
    <section>
      <div className={styles.gridFirstView} data-label="first-view">
        <Image src="/img/view/first-view.webp" alt="first-view" layout="fill" objectFit="cover" priority={true} />
        <div className={styles.textArea}>
          <div>
            <h2>Pandashark</h2>
            <p>Web Engineer</p>
            <NextArrowButton label="Contact" href="/contact/" onDark />
          </div>
        </div>
      </div>
    </section>
  );
};

export default FirstView;
