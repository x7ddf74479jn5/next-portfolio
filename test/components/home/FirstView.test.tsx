import FirstView from "src/components/home/FirstView";
import type { RenderResult } from "test/test-utils";
import { reTestCase } from "test/test-utils";
import { render } from "test/test-utils";

describe("FirstView", () => {
  let renderResult: RenderResult;

  beforeEach(() => {
    renderResult = render(<FirstView />);
  });

  it("matches snapshot", () => {
    expect(renderResult.asFragment()).toMatchSnapshot();
  });

  it("rendered correctly", () => {
    expect(renderResult.getByRole("img")).toHaveAttribute("src", reTestCase.anyWord);
    expect(renderResult.getByRole("img")).toHaveAttribute("alt", reTestCase.anyWord);

    expect(renderResult.getByRole("heading")).toHaveTextContent("Pandashark");

    expect(renderResult.getByRole("button")).toHaveTextContent("Contact");
    expect(renderResult.getByRole("button")).toHaveAttribute("href", expect.stringMatching("contact"));
  });
});
