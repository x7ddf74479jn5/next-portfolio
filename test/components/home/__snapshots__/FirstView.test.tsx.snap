// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FirstView matches snapshot 1`] = `
<DocumentFragment>
  <section>
    <div
      class="gridFirstView"
      data-label="first-view"
    >
      <div
        style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
      >
        <img
          alt="first-view"
          data-nimg="true"
          decoding="async"
          sizes="100vw"
          src="/_next/image?url=%2Fimg%2Fview%2Ffirst-view.webp&w=3840&q=75"
          srcset="/_next/image?url=%2Fimg%2Fview%2Ffirst-view.webp&w=640&q=75 640w, /_next/image?url=%2Fimg%2Fview%2Ffirst-view.webp&w=750&q=75 750w, /_next/image?url=%2Fimg%2Fview%2Ffirst-view.webp&w=828&q=75 828w, /_next/image?url=%2Fimg%2Fview%2Ffirst-view.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimg%2Fview%2Ffirst-view.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimg%2Fview%2Ffirst-view.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimg%2Fview%2Ffirst-view.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimg%2Fview%2Ffirst-view.webp&w=3840&q=75 3840w"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
        />
      </div>
      <div
        class="textArea"
      >
        <div>
          <h2>
            Pandashark
          </h2>
          <p>
            Web Engineer
          </p>
          <a
            class="btnRoundOnDark iconArrowNext"
            href="/contact"
            role="button"
          >
            Contact
          </a>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
